pkg/server/server.go

```diff
@@ -0,0 +1,118 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	database "github.com/sachintiptur/http-app/pkg/util"
+)
+
+// DatabaseInterfaceStruct holds the database interface type
+type DatabaseInterfaceStruct struct {
+	DatabaseInterface database.Database
+}
+
+// ValidateData Validate the data sent by client
+// Check for key and value length
+func ValidateData(key, val string) error {
+	if len(key) > database.MaxKeySz {
+		return fmt.Errorf("key length is greater than %d", database.MaxKeySz)
+	} else if len(val) > database.MaxValSz {
+		return fmt.Errorf("value length is greater than %d", database.MaxValSz)
+	}
+	return nil
+}
+
+// ProcessGET Process the HTTP GET request
+func ProcessGET(data database.Database, resp http.ResponseWriter, req *http.Request) {
+	key := req.URL.Query().Get("key")
+
+	// read the entry from database
+	value, err := data.Get(key)
+	if err != nil {
+		http.Error(resp, "Database entry not found", http.StatusNotFound)
+	} else {
+		resp.WriteHeader(http.StatusOK)
+		_, err = resp.Write([]byte(fmt.Sprintf("Data found for key %s: %s", key, value)))
+		if err != nil {
+			return
+		}
+	}
+}
+
+// ProcessPUT Process the HTTP PUT request
+func ProcessPUT(data database.Database, resp http.ResponseWriter, req *http.Request) {
```

Comment:
Wouldn't this be more interesting as DatabaseInterfaceStruct method? You wouldn't even need to pass the database as an argument.